### DISCUSSION TITLE
What's the actual difference between serialize and consume?

### DISCUSSION THREAD

**Discussion by lucasKuratani** [adonisjs/core#3199] [Help]
What is the difference between **serialize** and **consume** in the **@column.dateTime** model decorator?

They behave exactly the same, but their definitions differ.

**Consume's** description is defined as *Invoked when row is fetched from the database*, while **serialize** is described as *Invoked before serializing process happens*.

![image](https://user-images.githubusercontent.com/62770240/134520513-5c5f2df1-5a6a-494f-bbdc-9d5c106c0366.png)

**Comment by RomainLanz** [comment]
Hey @lucasKuratani! 👋🏻 

They do not behave the same.

Serialization happens when [you call `Model#serialize()`](https://docs.adonisjs.com/guides/models/serializing-models) (which is called automatically when request's end). You still use the raw database value inside your application.

The `consume` callback is called directly after fetching the row from your database, meaning it will be available as-is inside your application code.

Let's take a date as an example. You may want to use a `Luxon` instance inside your application, but you want the value stringified in a specific format when sending the response (serializing).
